Kernel/Storage/IntelAHCIController.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021, Alexander Richards <electrodeyt@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Format.h>
+#include <Kernel/CommandLine.h>
+#include <Kernel/Storage/IntelAHCIController.h>
+#include <Kernel/Storage/IntelAHCIPort.h>
+
+namespace Kernel {
+
+NonnullRefPtr<AHCIController> IntelAHCIController::initialize(PCI::Address address)
+{
+    return adopt_ref(*new IntelAHCIController(address));
+}
+
+IntelAHCIController::IntelAHCIController(PCI::Address address)
+    : AHCIController(address, false)
+{
+    initialize();
+}
+
+void IntelAHCIController::initialize()
```

Comment:
This feels like a duplication of `AHCIController::initialize` method. Maybe we can do it in a better way, like do specific stuff for the Intel AHCI controller and then call the more generic `AHCIController::initialize` method?